tests/test_sequence.py

```diff
@@ -137,6 +138,327 @@ def test_magnetic_field():
+        match="Switching the device of a sequence to one"
+        + " with reusable channels is not supported.",
+    ):
+        seq.switch_device(MockDevice)
+
+    seq = init_seq(MockDevice, "ising", "rydberg_global", None)
+    with pytest.raises(
+        ValueError,
+        match="Device match failed because the devices"
+        + " have different Rydberg levels.",
+    ):
+        seq.switch_device(IroiseMVP, True)
+
+    # Different Channels basis
+    seq = init_seq(devices[0], "ising", "raman_global", None)
+    with pytest.raises(
+        TypeError,
+        match="No match for channel ising with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(Chadoq2)
+
+    # Different addressing channels
+
+    with pytest.raises(
+        TypeError,
+        match="No match for channel ising with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(devices[1])
+
+    # Strict: Jump_phase_time & CLock-period criteria
+    # Jump_phase_time check 1: phase not nill
+
+    seq = init_seq(
+        devices[2],
+        channel_name="ising",
+        channel_id="rydberg_global",
+        l_pulses=pulses[:2],
+    )
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right phase_jump_time & clock_period.",
+    ):
+        seq.switch_device(MockDevice, True)
+
+    # Jump_phase_time check 2: No phase
+
+    seq = init_seq(
+        devices[2],
+        channel_name="ising",
+        channel_id="rydberg_global",
+        l_pulses=[pulses[0], pulses[2]],
+    )
+    with pytest.warns(
+        UserWarning,
+        match="The phase_jump_time of the matching channel"
+        + " on the the new device is different, take it into account"
+        + " for the upcoming pulses.",
+    ):
+        seq.switch_device(Chadoq2, True)
+
+    # Clock_period not match
+    seq = init_seq(
+        devices[0],
+        channel_name="ising",
+        channel_id="rydberg_global",
+        l_pulses=pulses[:2],
+    )
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right phase_jump_time & clock_period.",
+    ):
+        seq.switch_device(devices[1], True)
+
+    seq = init_seq(
+        devices[2],
+        channel_name="digital",
+        channel_id="rmn_local1",
+        l_pulses=[],
+        initial_target=["q0"],
+    )
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel digital"
+        + " with the right mod_bandwidth.",
+    ):
+        seq.switch_device(devices[0], True)
+
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel digital"
+        + " with the right fixed_retarget_t.",
+    ):
+        seq.switch_device(devices[1], True)
+
+
+@pytest.mark.parametrized("device_ind, strict", [(1, False), (2, True)])
+def test_switch_device_up(devices, pulses, device_ind, strict):
```

Comment:
The test might have vailed because of the order of the arguments, maybe try 
```suggestion
def test_switch_device_up(device_ind, strict, devices, pulses):
```
Test it locally first and the push, rather than just committing my suggestion.

Reply:
Why is it taking `device_ind` as a fixture?

Reply:
It's `..._parametrize` instead of `..._parametrized`

Reply:
Oh okay, good catch!